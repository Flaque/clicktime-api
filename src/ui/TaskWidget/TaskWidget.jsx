import React from 'react';
import Search from '../search/Search.jsx'
import DisplayView from '../displayView/DisplayView.jsx'
import Loader from '../loader/Loader.jsx'
import {getAllTasks} from '../../network/ClickTimeAPI/ClickTimeAPI';

/**
 * Creates an array of {name, id} pairs from the
 * task data object
 *
 * @return array
 */
const _getNameIdPairs = function(taskDict) {
  let pairs = []
  for (let id in taskDict) {
    pairs.push({
      name: taskDict[id].Name,
      id: id
    })
  }
  return pairs
}

class TaskWidget extends React.Component {

  /**
   * Loads Tasks asynchronously and then updates the state.
   */
  load() {
    const creds = this.props.credentials
    getAllTasks(creds.companyID, creds.userID)
      .then( (data) => {
        this.setState({
          tasks: data,
          namePairs: _getNameIdPairs(data),
          isLoaded: true
        })
       })
  }

  /**
   * Selects a specific task
   */
  selectTask(id) {
    this.setState({ selectedTask: this.state.tasks[id] })
  }

  constructor(props) {
    super(props)
    this.state = { selectedTask: {}, tasks: {}, namePairs: [] }
    this.load()
  }

  render() {
    return (
      <div className="taskWidget container">

        {this.state.isLoaded ? (

          <div>
            <div className="header">
              <h4> Tasks </h4>
              <Search items={ this.state.namePairs }
                onSelect={ this.selectTask.bind(this) } />
            </div>

            <div className="card">
              <DisplayView selectedTask={ this.state.selectedTask } />
            </div>
          </div>
        ) : (
          <Loader loaderMessage="Logging in..."/>
        )}
      </div>
    )
  }
}

export default TaskWidget;
