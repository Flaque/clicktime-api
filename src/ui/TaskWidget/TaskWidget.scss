@import '../../lib/animations.scss';


.taskWidget {

  .card {
    @include fadein(1.5s);
  }

  .header {
    height: 50px;
    @include fadein(1.5s);

    h1,h2,h3,h4,h5,h6 {
      float: left;
      color: white;
    }

    .search {
      float: right;
    }

    input {
      border: none;

      &.card {
        border: none;

        &:focus {
          border: none;
        }
      }
    }
  }
}
