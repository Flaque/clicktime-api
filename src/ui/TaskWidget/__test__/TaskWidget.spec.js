import React from 'react';
import TaskWidget from '../TaskWidget.jsx';
import renderer from 'react-test-renderer';

const mockCompanyID = '2UKgcxIb17NY';
const mockUserID = '2i86jjewbXL4';
const mockCredentials = {
  companyID : mockCompanyID,
  userID : mockUserID
}

test('Test that TaskWidget is still the same', () => {
  const tree = renderer.create(
    <TaskWidget credentials={mockCredentials}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
