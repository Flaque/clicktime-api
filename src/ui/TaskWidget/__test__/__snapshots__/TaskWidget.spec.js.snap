exports[`test Test that TaskWidget is still the same 1`] = `
<div
  className="taskWidget container">
  <div
    id="loader">
    <div
      className="loader-content">
      <h4>
        Logging in...
      </h4>
      <div
        className="nob pulse" />
    </div>
  </div>
</div>
`;
